show-line-of-sight-between-geoelements/src/main/java/com/esri/arcgismaps/sample/showlineofsightbetweengeoelements/components/SceneViewModel.kt

```diff
@@ -0,0 +1,274 @@
+    }
+
+    // Create a graphic near the Empire State Building to be the observer
+    private val observerGraphic = Graphic(
+        geometry = Point(
+            x = -73.9853,
+            y = 40.7484,
+            z = 50.0,
+            spatialReference = SpatialReference.wgs84()
+        ),
+        symbol = SimpleMarkerSymbol(
+            style = SimpleMarkerSymbolStyle.Circle,
+            color = Color.red,
+            size = 5f
+        )
+    )
+
+    // Zoom to show the observer
+    private val camera = Camera(
+        lookAtPoint = observerGraphic.geometry as Point,
+        distance = 700.0,
+        roll = 0.0,
+        pitch = 45.0,
+        heading = -30.0,
+    )
+
+
+    // Define base surface for elevation data
+    private val surface = Surface().apply {
+        elevationSources.add(
+            ArcGISTiledElevationSource(
+                uri = getString(
+                    application,
+                    R.string.elevation_service_url
+                )
+            )
+        )
+    }
+
+    // Define a scene layer for the New York buildings
+    private val buildings =
+        ArcGISSceneLayer(uri = application.getString(R.string.new_york_buildings_service_url))
+
+
+    // Create a scene and add a basemap to it.
+    // Set the surface and buildings in the scene, and define the viewpoint on launch
+    val scene = ArcGISScene(BasemapStyle.ArcGISTopographic).apply {
+        baseSurface = surface
+        operationalLayers.add(buildings)
+        initialViewpoint = Viewpoint(
+            boundingGeometry = observerGraphic.geometry as Point,
+            camera = camera
+        )
+    }
+
+
+    // Set up a heading expression to handle graphic rotation
+    private val renderer3D = SimpleRenderer().apply {
+        sceneProperties.headingExpression = ("[HEADING]")
+    }
+
+
+    // Create graphic overlay to hold graphics
+    // Set the surface placement, renderer, and add graphics,
+    val graphicsOverlay = GraphicsOverlay().apply {
+        sceneProperties.surfacePlacement = SurfacePlacement.RelativeToScene
+        renderer = renderer3D
+        graphics.addAll(listOf(observerGraphic, taxiGraphic))
+    }
+
+
+    // Create a line of sight between the two graphics and add it to the analysis overlay
+    private val lineOfSight = GeoElementLineOfSight(
+        observerGeoElement = observerGraphic,
+        targetGeoElement = taxiGraphic
+    ).apply {
+        // Observe the visibility status of the moving taxi
+        viewModelScope.launch(Dispatchers.Main) {
+            targetVisibility.collect { targetVisibility ->
+                updateTargetVisibilityString(
+                    when (targetVisibility) {
+                        is LineOfSightTargetVisibility.Visible -> "Visible"
+                        is LineOfSightTargetVisibility.Obstructed -> "Obstructed"
+                        is LineOfSightTargetVisibility.Unknown -> "Unknown"
+                    }
+                )
```

Comment:
```suggestion
                // Update visibility string and select/highlight the taxi based on visibility status
                when (targetVisibility) {
                    is LineOfSightTargetVisibility.Visible -> {
                        updateTargetVisibilityString("Visible")
                        taxiGraphic.isSelected = true
                    }
                    is LineOfSightTargetVisibility.Obstructed -> {
                        updateTargetVisibilityString("Obstructed")
                        taxiGraphic.isSelected = false
                    }
                    is LineOfSightTargetVisibility.Unknown -> {
                        updateTargetVisibilityString("Unknown")
                        taxiGraphic.isSelected = false
                    }
                }
```

This is one way to update the taxi selection, by putting it in the collector of targetVisibility flow